sdk/src/asset_handlers/jpeg_io.rs

```diff
@@ -574,11 +575,53 @@ impl RemoteRefEmbed for JpegIO {
 
     fn embed_reference_to_stream(
         &self,
-        _source_stream: &mut dyn CAIRead,
-        _output_stream: &mut dyn CAIReadWrite,
-        _embed_ref: RemoteRefEmbedType,
+        source_stream: &mut dyn CAIRead,
+        output_stream: &mut dyn CAIReadWrite,
+        embed_ref: RemoteRefEmbedType,
     ) -> Result<()> {
-        Err(Error::UnsupportedType)
+        match embed_ref {
+            crate::asset_io::RemoteRefEmbedType::Xmp(manifest_uri) => {
+                pub const MIN_XMP: &str = r#"<?xpacket begin="" id="W5M0MpCehiHzreSzNTczkc9d"?> <x:xmpmeta xmlns:x="adobe:ns:meta/" x:xmptk="XMP Core 6.0.0"><rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"><rdf:Description rdf:about="" > </rdf:Description></rdf:RDF> </x:xmpmeta> "#;
+                let mut buf = Vec::new();
+                // read the whole asset
+                source_stream.rewind()?;
+                source_stream
+                    .read_to_end(&mut buf)
+                    .map_err(Error::IoError)?;
+                let mut jpeg =
+                    Jpeg::from_bytes(buf.into()).map_err(|_err| Error::EmbeddingError)?;
+
+                // first extract the xmp from APP1 markers
+                let app1_segs = jpeg.segments_by_marker(markers::APP1);
+                let mut xmp: String = app1_segs.filter_map(extract_xmp).collect();
+
+                // remove existing XMP segments
+                let segments = jpeg.segments_mut();
+                segments.retain(|seg| {
+                    !(seg.marker() == markers::APP1 && seg.contents().starts_with(XMP_SIGNATURE))
+                });
+
+                if xmp.is_empty() {
+                    xmp = format!("http://ns.adobe.com/xap/1.0/\0 {}", MIN_XMP);
```

Comment:
What is the purpose of this line? It looks like `http://ns.adobe.com/xap/1.0/\0` exists outside of the XMP?